diff --git a/include/env_default.h b/include/env_default.h
index 853c290042..cf2e9e256b 100644
--- a/include/env_default.h
+++ b/include/env_default.h
@@ -10,6 +10,8 @@
 
 #include <env_callback.h>
 
+#include <env_resin.h>
+
 #ifdef DEFAULT_ENV_INSTANCE_EMBEDDED
 env_t environment __UBOOT_ENV_SECTION__ = {
 	ENV_CRC,	/* CRC Sum */
@@ -22,6 +24,7 @@ static char default_environment[] = {
 #else
 const uchar default_environment[] = {
 #endif
+	BALENA_ENV
 #ifdef	CONFIG_ENV_CALLBACK_LIST_DEFAULT
 	ENV_CALLBACK_VAR "=" CONFIG_ENV_CALLBACK_LIST_DEFAULT "\0"
 #endif
